Convert PeriodicTablePage to function component

sudokoi opened the pull request:
Closes #115

sudokoi committed:
Convert PeriodicTablePage to function component

Refer: #115

mpeyper commented on src/components/periodic-table-page/PeriodicTablePage.tsx:
```diff
@@ -10,102 +8,90 @@ import PeriodicTable, {
 
-@autobind
-class PeriodicTablePage extends React.Component<
-  Props,
-  IPeriodicTablePageState
-> {
-  public state: IPeriodicTablePageState = {
+function PeriodicTablePage() {
+  const history = useHistory();
+
+  const [state, setState] = React.useState<IPeriodicTablePageState>(() => ({
     elementInfo: {
       element: ElementManager.getElement(1) as IElement,
       open: false,
     },
-  };
```
generally we would name the state when using `useState` instead of just using `state` and `setState`.  In this case, I don't thing the `elementInfo` layer of the state is actually giving us anything so I suggest using that:

```ts
  const [elementInfo, setElementInfo] = React.useState<IPeriodicTablePageState>(() => ({
    element: ElementManager.getElement(1) as IElement,
    open: false,
  }));
```

mpeyper commented on src/components/periodic-table-page/PeriodicTablePage.tsx:
```diff
@@ -10,102 +8,90 @@ import PeriodicTable, {
 
-type Props = RouteComponentProps<any> & React.Props<any>;
-
 interface IPeriodicTablePageState {
```
Not necessary, but the repo owner has stated in other PRs that he would prefer the `I...` prefix on interfaces be dropped where we're making changes.

mpeyper commented on src/components/periodic-table-page/PeriodicTablePage.tsx:
```diff
@@ -10,102 +8,90 @@ import PeriodicTable, {
-        >
-          <ElementInfo element={elementInfo.element} />
-        </SwipeableModal>
-      </div>
-    );
-  }
+  const onNavbarBackButtonClick = () => {
+    history.push(MAIN_MENU);
+  };
 
-  private elementRenderer(atomic: number): IPeriodicTableElement {
+  const elementRenderer = (atomic: number): IPeriodicTableElement => {
     return {
       // @ts-ignore fix this
       component: PtElementInfo,
       props: {
         element: ElementManager.getElement(atomic),
-        onClick: this.elementOnClick,
+        onClick: (element: IElement) => {
+          openElementInfo(element);
+        },
       },
     };
-  }
```
You may want to sneak a peek at what's happening in [this PR (#111)](https://github.com/OpenSourceRaidGuild/atom-pwa/pull/111/files#diff-03058fb889f1b9366517962ee8952715a92ea027c14176bdd61eaed3af066fa2)

sudokoi replied:
Okay. I see. This has already been conquered. I shall close my pr then.

mpeyper replied:
I don't think it has, just the elementRenderer interface changing (if the PR is merged)

sudokoi replied:
Let that PR merge first..I will pull the changes and then open this PR for review. Thanks for the heads up!

mpeyper reviewed (changes requested):
I've marked as "request changes" but I'm not willing to die on any of the hills here, so feel free to ignore them.

sudokoi committed:
pull changes from upstream

sudokoi committed:
Make suggested changes:

 - drop the `I` from interface name
 - replace the [state, setState] with more semantic names

sudokoi committed:
Merge branch 'next' of https://github.com/OpenSourceRaidGuild/atom-pwa into next

sudokoi committed:
merge changes from upstream

mpeyper reviewed (approved):
Thanks for the changes

sudokoi commented:
> Thanks for the changes

You don't really need to thank me. Thank you for reviewing my code! 😊
